XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -197,6 +205,25 @@ private void CreateCHIN()
             (new CreateChinAction(xenConnection, network, theInterface)).RunAsync();
         }
 
+        private void CreateSRIOV()
+        {
+            XenAPI.Network network = PopulateNewNetworkObj();
+
+            List<PIF> sriovSelectedPifs = new List<PIF>();
+            Pool pool = Helpers.GetPoolOfOne(Host.Connection);
+            foreach (Host host in pool.Connection.Cache.Hosts)
+            {
+                foreach (PIF thePIF in Host.Connection.ResolveAll<PIF>(host.PIFs))
+                {
+                    if (thePIF.IsPhysical() && !thePIF.IsBondNIC() && thePIF.SriovCapable() && thePIF.Name() == pageSriovDetails.SelectedHostNic.Name() && thePIF.sriov_physical_PIF_of.Count == 0)
+                        sriovSelectedPifs.Add(thePIF);
+                }
+            }
+
+            CreateSriovAction action = new CreateSriovAction(xenConnection, network, sriovSelectedPifs);

```

Comment:
Shall we continue with running the action here if the sriovSelectedPifs is empty? Because in this case the action will only create a new network object, without any network-sriov objects.
Actually, should we even allow the users to select a NIC that is already the physical NIC for a network_sriov? (on the NetWSriovDetails page)

Reply:
OK, I have filtered out the NIC that has enable SR-IOV on the NetWSriovDetails page. Thanks.